homeassistant/components/insteon/climate.py

```diff
@@ -0,0 +1,228 @@
+            return self._insteon_device.groups[HEAT_SET_POINT].value
+        return None
+
+    @property
+    def fan_mode(self) -> Optional[str]:
+        """Return the fan setting."""
+        return FAN_MODES[self._insteon_device.groups[FAN_MODE].value]
+
+    @property
+    def fan_modes(self) -> Optional[List[str]]:
+        """Return the list of available fan modes."""
+        return list(FAN_MODES.values())
+
+    @property
+    def target_humidity(self) -> Optional[int]:
+        """Return the humidity we try to reach."""
+        high = self._insteon_device.groups[HUMIDITY_HIGH].value
+        low = self._insteon_device.groups[HUMIDITY_LOW].value
+        # May not be loaded yet so return a default if required
+        return (high + low) / 2 if high and low else None
+
+    @property
+    def min_humidity(self) -> int:
+        """Return the minimum humidity."""
+        return 1
+
+    @property
+    def hvac_action(self) -> Optional[str]:
+        """Return the current running hvac operation if supported.
+
+        Need to be one of CURRENT_HVAC_*.
+        """
+        if self._insteon_device.groups[COOLING].value:
+            return CURRENT_HVAC_COOL
+        if self._insteon_device.groups[HEATING].value:
+            return CURRENT_HVAC_HEAT
+        if self._insteon_device.groups[FAN_MODE].value == ThermostatMode.FAN_ALWAYS_ON:
+            return CURRENT_HVAC_FAN
+        return CURRENT_HVAC_IDLE
+
+    @property
+    def device_state_attributes(self):
+        """Provide attributes for display on device card."""
+        attr = super().device_state_attributes
+        humidifier = "off"
+        if self._insteon_device.groups[DEHUMIDIFYING].value:
+            humidifier = "dehumidifying"
+        if self._insteon_device.groups[HUMIDIFYING].value:
+            humidifier = "humidifying"
+        attr["humidifier"] = humidifier
+        return attr
+
+    async def async_set_temperature(self, **kwargs) -> None:
+        """Set new target temperature."""
+        target_temp = kwargs.get(ATTR_TEMPERATURE)
+        target_temp_low = kwargs.get(ATTR_TARGET_TEMP_LOW)
+        target_temp_high = kwargs.get(ATTR_TARGET_TEMP_HIGH)
+        if target_temp is not None:
+            if self._insteon_device.groups[SYSTEM_MODE].value == ThermostatMode.HEAT:
+                await self._insteon_device.async_set_heat_set_point(target_temp)
+            elif self._insteon_device.groups[SYSTEM_MODE].value == ThermostatMode.COOL:
+                await self._insteon_device.async_set_cool_set_point(target_temp)
+        else:
+            await self._insteon_device.async_set_heat_set_point(target_temp_low)
+            await self._insteon_device.async_set_cool_set_point(target_temp_high)
+
+    async def async_set_fan_mode(self, fan_mode: str) -> None:
+        """Set new target fan mode."""
+        mode = list(FAN_MODES.keys())[list(FAN_MODES.values()).index(fan_mode)]
```

Comment:
It would be easier to define another constant that is the inverse of the `FAN_MODES` constant. Then just look up that dict here. Probably rename `FAN_MODES` too.
```py
FAN_MODES_HA_INSTEON = {v: k for k, v in FAN_MODES_INSTEON_HA.items()}
```